jms/1.7/modules/ROOT/pages/jms-listener.adoc

```diff
@@ -46,66 +79,79 @@ For example, if you have a priority Queue with Messages that need to be processe
+When an incoming JMS message declares a `ReplyTo` header destination and the message is processed successfully without any error during the flow execution, the *On New Message* source automatically produces a response.
+In that case, when the flow completes the connector publishes the response to the destination specified in the *Destination* field.
 
+In the following example, you configure the response message section of the operation. All these configurations impact the outbound message:
 
-== Replying to Incoming Messages
+. In Studio, select the *On New Message* source from your flow.
+. In the *Response* section, set the *Ignore Jms replyTo Header* to *True*, which disables the automatic reply.
+. Set the *Body* field to `'Message received was: ' ++ payload`.
+. Set the *User Properties* field to `{'processedAt': now}`.
+. Set the *Persistent delivery* to *True*.
+. Set the *Priority* field to `8`.
 
-When an incoming JMS Message declares a REPLY_TO destination, the JMS Listener will automatically produce a response *when the Message is processed successfully*, meaning that no error occurs during the flow execution.
-In that case, when the flow is completed a response will be published to the destination specified in the processed Message header.
+.JMS Connector On New Message Response configuration
+image::jms-onnewmessage-response.png[In the On New Message source screen, configure the Response section fields]
 
-Responses can be configured in the Listener with the following syntax:
+In the XML editor, the `ignoreReplyTo`, `body`, `properties`, `persistentDelivery`, and `priority` configurations looks like this:
 
 [source,xml,linenums]
 ----
 <jms:listener config-ref="config" destination="#[vars.destination]">
-  <jms:response priority="8" persistentDelivery="true">
+  <jms:response priority="8" persistentDelivery="true" ignoreReplyTo="true">
       <jms:body>#['Message received was: ' ++ payload]</jms:body>
       <jms:properties>#[{'processedAt': now}]</jms:properties>
   </jms:response>
 </jms:listener>
 ----
 
-This response builder allows you to configure not only the `body` and `properties` of the response message but also allows you to specify any header like `priority` and `persistentDelivery`.
-All these configurations impact in the outbound Message and can be built using all the information available in the flow when the processing is completed.
+== Configure the Number of Consumers
 
-Disable the automatic reply by using the `ignoreReplyTo` attribute in the `jms:response` element:
+When extra processing power is needed, configure in the *Number of consumers* field the number of consumers that the *On New Message* source uses to consume messages concurrently. +
+By default, each source uses `4` consumers that produce messages concurrently. Each consumer waits for the message to be processed, which means that you have a maximum of `4` messages in-flight at the same time.
```

Comment:
```suggestion
By default, each source is configured to enable `4` consumers to produce messages concurrently. Each consumer waits for the message to be processed, which means that you have a maximum of `4` messages being produced at the same time.
```
I'm not sure that I understand "Each consumer waits for the message to be processed." Does this mean that each consumer waits for all four messages to be processed? Waiting for what; what is being delayed by the wait?